tests/behat/src/CmsFormsContext.php

```diff
@@ -445,6 +446,23 @@ public function stepIClickTheGridFieldButtonForRow($buttonLabel, $gridFieldName,
         $button->click();
     }
 
+    /**
+     * @When /^I click the "([^"]*)" button in the "([^"]*)" gridfield for the "([^"]*)" row, confirming the dialog$/
+     * @param string $buttonLabel
+     * @param string $gridFieldName
+     * @param string $rowName
+     */
+    public function stepIClickTheGridFieldButtonForRowConfirmingDialog($buttonLabel, $gridFieldName, $rowName)
+    {
+        $this->stepIClickTheGridFieldButtonForRow($buttonLabel, $gridFieldName, $rowName);
+        $session = $this->getSession()->getDriver()->getWebDriver();
+        $session->wait()->until(
+            WebDriverExpectedCondition::alertIsPresent(),
+            "Alert is expected"
+        );
+        $session->switchTo()->alert()->accept();
+    }
```

Comment:
There's a version of this (see `stepIClickTheGridFieldButtonForRow`) that doesn't confirm a dialog - that's fine for some actions, like "edit". The delete action pops up a dialog though, and you can't confirm the dialog in a separate step because behat won't run that step - it halts as soon as it sees the dialog box.